### PR TITLE
Fix Aplib decompression

### PR DESCRIPTION
**Pull request by sisoma2**
Using the function malduck.aplib() returned a None object.

We can see that running the tests:
image

After the fixes in this PR the tests on compression are passed:
image

### CODE REVIEW

File: malduck/compression/components/aplib.py [hunk 1]
```diff
@@ -1,132 +1,172 @@
...
+        raise NotImplementedError
 
-                    r0 = offs
-                    lwm = 1
-            else:
-                offs = ap_getgamma(ud)
 
-                if lwm == 0 and offs == 2:
-                    offs = r0
+def decompress(data, strict=False):
```
Comment (nazywam) [inline malduck/compression/components/aplib.py:137]: I think this part (+ main) is duplicated with `compression/aplib.py` and we don't need it anymore here?

**Review by nazywam — COMMENTED** [review]
Thanks! Left a few comments

File: malduck/compression/components/aplib.py (hunk 1 above)
Comment (sisoma2) [reply malduck/compression/components/aplib.py]: Deleted the decompress function from aplib.py

**Comment by sisoma2** [comment]
Thanks for the review! All the suggested changes are fixed and committed 👍

**Comment by nazywam** [comment]
This diff fixes the pipeline for me:
```diff
diff --git a/malduck/compression/aplib.py b/malduck/compression/aplib.py
index 8e75303..39efe4b 100644
--- a/malduck/compression/aplib.py
+++ b/malduck/compression/aplib.py
@@ -1,4 +1,5 @@
 from typing import Optional
+from binascii import crc32
 
 from .components.aplib import APLib
 
@@ -37,24 +38,30 @@ class aPLib:
         orig_crc = None
         strict = not headerless
 
-        if buf.startswith(b'AP32') and len(buf) >= 24:
+        if buf.startswith(b"AP32") and len(buf) >= 24:
             # buf has an aPLib header
-            header_size, packed_size, packed_crc, orig_size, orig_crc = struct.unpack_from('=IIIII', buf, 4)
+            (
+                header_size,
+                packed_size,
+                packed_crc,
+                orig_size,
+                orig_crc,
+            ) = struct.unpack_from("=IIIII", buf, 4)
             buf = buf[header_size : header_size + packed_size]
 
         if strict:
             if packed_size is not None and packed_size != len(buf):
-                raise RuntimeError('Packed buf size is incorrect')
+                raise RuntimeError("Packed buf size is incorrect")
             if packed_crc is not None and packed_crc != crc32(buf):
-                raise RuntimeError('Packed buf checksum is incorrect')
+                raise RuntimeError("Packed buf checksum is incorrect")
 
         result = APLib(buf, strict=strict).depack()
 
         if strict:
             if orig_size is not None and orig_size != len(result):
-                raise RuntimeError('Unpacked buf size is incorrect')
+                raise RuntimeError("Unpacked buf size is incorrect")
             if orig_crc is not None and orig_crc != crc32(result):
-                raise RuntimeError('Unpacked buf checksum is incorrect')
+                raise RuntimeError("Unpacked buf checksum is incorrect")
 
         return result
 
diff --git a/malduck/compression/components/aplib.py b/malduck/compression/components/aplib.py
index 267bbb0..9c010e6 100644
--- a/malduck/compression/components/aplib.py
+++ b/malduck/compression/components/aplib.py
@@ -5,18 +5,16 @@ Adapted from the original C source code from http://ibsensoftware.com/files/aPLi
 Approximately 20 times faster than other Python implementations.
 Compatible with both Python 2 and 3.
 """
-import struct
-from binascii import crc32
 from io import BytesIO
 
-__all__ = ['APLib', 'decompress']
-__version__ = '0.6'
-__author__ = 'Sandor Nemes'
+__all__ = ["APLib"]
+__version__ = "0.6"
+__author__ = "Sandor Nemes"
 
 
 class APLib(object):
 
-    __slots__ = 'source', 'destination', 'tag', 'bitcount', 'strict'
+    __slots__ = "source", "destination", "tag", "bitcount", "strict"
 
     def __init__(self, source: bytes, strict: bool = True) -> None:
         self.source = BytesIO(source)
@@ -126,18 +124,9 @@ class APLib(object):
 
         except (TypeError, IndexError):
             if self.strict:
-                raise RuntimeError('aPLib decompression error')
+                raise RuntimeError("aPLib decompression error")
 
         return bytes(self.destination)
 
     def pack(self):
         raise NotImplementedError
-
-def main():
-    # self-test
-    data = b'T\x00he quick\xecb\x0erown\xcef\xaex\x80jumps\xed\xe4veur`t?lazy\xead\xfeg\xc0\x00'
-    assert decompress(data) == b'The quick brown fox jumps over the lazy dog'
-
-
-if __name__ == '__main__':
-    main()
diff --git a/tests/test_compression.py b/tests/test_compression.py
index f256264..573203f 100644
--- a/tests/test_compression.py
+++ b/tests/test_compression.py
@@ -22,8 +22,6 @@ QacB19//yAF9ff/8hwHX3//IAX19//yHAdff/8gBfX3//IcB19//yAF9ff/
 8hwHX3//IAX19//yH\nAdff/8gBXXf/2QqAAA==
 """)) == b"A"*1024*1024 + b"\n"
 
-    assert aplib(b"helloworld") is None
-    assert aplib(b"") is None
     assert (
         aplib(b'T\x00he quick\xecb\x0erown\xcef\xaex\x80jumps\xed\xe4veur`t?lazy\xead\xfeg\xc0\x00') ==
         b'The quick brown fox jumps over the lazy dog')

```

**Review by nazywam — APPROVED** [review]
Looks good, thanks! Sorry about the hassle with CI

**Comment by sisoma2** [comment]
No problem Haha Thanks to you for the quick review and merge. Do you plan to publish a new Pypi release for the other projects that use the package?

**Pull request merged**